### PR TITLE
Automatically compute migrations to execute

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

1. New migrations should be automatically taken into account (even if now we will have to update the list in tests).
2. Replaying latest migration now requires almost no specific code.
3. Files are naturally sorted.

### CODE REVIEW

File: tests/functionnal/Update.php [hunk 1]
```diff
@@ -83,4 +83,120 @@ function () use (&$migration) {
...
+            // Force latests does not duplicate latest in list
+            'current_version'     => '10.0.0-dev',
+            'force_latest'        => true,
+            'expected_migrations' => [
+               $path . '/update_9.5.x_to_10.0.0.php' => 'update95xto1000',
+            ],
+         ],
+         [
+            // Validate that list is empty when version matches
+            'current_version'     => '10.0.0',
```
Comment (trasher) [inline tests/functionnal/Update.php:178]:
```suggestion
            'current_version'     => GLPI_VERSION,
```

Comment (trasher) [reply tests/functionnal/Update.php:178]: indeed

File: inc/update.class.php [hunk 2]
```diff
@@ -468,4 +257,55 @@ public function getExpectedSecurityKeyFilePath(): ?string {
...
+      $migration_iterator = new DirectoryIterator(GLPI_ROOT . '/install/migrations/');
+      foreach ($migration_iterator as $file) {
+         $versions_matches = [];
+         if ($file->isDir() || $file->isDot() || preg_match($pattern, $file->getFilename(), $versions_matches) !== 1) {
+            continue;
+         }
+
+         $force_migration = false;
+         if ($current_version === '9.2.2' && $versions_matches['target_version'] === '9.2.2') {
+            //9.2.2 upgrade script was not run from the release, see https://github.com/glpi-project/glpi/issues/3659
```
Comment (trasher) [inline inc/update.class.php:292]:
```suggestion
            //9.2.2 upgrade script was not run from the release,
```

**Review by trasher — APPROVED** [review]

File: inc/update.class.php (hunk 2 above)
Comment (trasher) [reply inc/update.class.php:292]: is also present on next line

**Review by orthagh — APPROVED** [review]

**Pull request merged**